benchmark/single-source/SetTests.swift

```diff
@@ -12,107 +12,288 @@
 
 import TestsUtils
 
+let size = 400
+let half = size / 2
+let quarter = size / 4
+
+// Construction kit for sets with 25% overlap
+let setAB = Set(0 ..< size)                              //   0 ..< 400
+let setCD = Set(size ..< 2 * size)                       // 400 ..< 800
+let setBC = Set(size - quarter ..< 2 * size - quarter)   // 300 ..< 700
+let setB = Set(size - quarter ..< size)                  // 300 ..< 400
+
+let setOAB = Set(setAB.map(Box.init))
+let setOCD = Set(setCD.map(Box.init))
+let setOBC = Set(setBC.map(Box.init))
+let setOB = Set(setB.map(Box.init))
+
+let countA = size - quarter        // 300
+let countB = quarter               // 100
+let countC = countA                // 300
+let countD = countB                // 100
+
+let countAB = size                 // 400
+let countAC = countA + countC      // 600
+let countABC = countA + countB + countC           // 700
+let countABCD = countA + countB + countC + countD // 800
+
+// Construction kit for sets with 50% overlap
+let setXY = Set(0 ..< size)           //   0 ..< 400
+let setYZ = Set(half ..< size + half) // 200 ..< 600
+let setY = Set(half ..< size)         // 200 ..< 400
+
+// Two sets with 100% overlap, but different bucket counts (let's not make it
+// too easy...)
+let setP = Set(0 ..< size)
+let setQ: Set<Int> = {
+  var set = Set(0 ..< size)
+  set.reserveCapacity(2 * size)
+  return set
+}()
+
 public let SetTests = [
-  BenchmarkInfo(name: "SetExclusiveOr", runFunction: run_SetExclusiveOr, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetExclusiveOr_OfObjects", runFunction: run_SetExclusiveOr_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIntersect", runFunction: run_SetIntersect, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIntersect_OfObjects", runFunction: run_SetIntersect_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIsSubsetOf", runFunction: run_SetIsSubsetOf, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetIsSubsetOf_OfObjects", runFunction: run_SetIsSubsetOf_OfObjects, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetUnion", runFunction: run_SetUnion, tags: [.validation, .api, .Set]),
-  BenchmarkInfo(name: "SetUnion_OfObjects", runFunction: run_SetUnion_OfObjects, tags: [.validation, .api, .Set]),
+  // Mnemonic: number after name is percentage of common elements in input sets.
+  BenchmarkInfo(
+    name: "SetIsSubsetInt0",
+    runFunction: { n in run_SetIsSubsetInt(setAB, setCD, false, 5000 * n) },
+    tags: [.validation, .api, .Set],
```

Comment:
Tags are same for all test here. How about extracting this to constant and DRY the declarations a bit?

`name: "Set…", tags: tags,`